harness/variables.go

```diff
@@ -0,0 +1,191 @@
+		return err
+	}
+
+	// Write the error category
+	if err := UpdateOrRemoveKeyValue(MetadataFile, ErrorCategoryKey, category, false); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateOrRemoveKeyValue updates or deletes a key-value pair in the specified file.
+func UpdateOrRemoveKeyValue(envVar, key, newValue string, delete bool) error {
+	// Get the file path from the environment variable
+	filePath := os.Getenv(envVar)
+	if filePath == "" {
+		return fmt.Errorf("environment variable %s is not set", envVar)
+	}
+
+	// Ensure the file exists before reading
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		// Create the file if it does not exist
+		_, err := os.OpenFile(filePath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+		if err != nil {
+			return fmt.Errorf("failed to create file: %w", err)
+		}
+	}
+
+	// Determine the file extension to handle formats
+	ext := strings.ToLower(filepath.Ext(filePath))
+
+	// Read the file contents into memory
+	lines, err := ReadLines(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to read file: %w", err)
+	}
+
+	// Process lines
+	var updatedLines []string
+	found := false
+	for _, line := range lines {
+		k, v := ParseKeyValue(line, ext)
+		if k == key {
+			found = true
+			if delete {
+				continue // Skip the line to delete it
+			}
+			updatedLines = append(updatedLines, FormatKeyValue(k, newValue, ext))
+		} else {
+			updatedLines = append(updatedLines, FormatKeyValue(k, v, ext))
+		}
+	}
+
+	// Append new key-value if not found and not deleting
+	if !found && !delete {
+		updatedLines = append(updatedLines, FormatKeyValue(key, newValue, ext))
+	}
+
+	// Write updated lines back to the file
+	return WriteLines(filePath, updatedLines)
+}
+
+// ReadLines reads lines from a file and returns them as a slice of strings.
+func ReadLines(filename string) ([]string, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	var lines []string
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+	return lines, scanner.Err()
+}
+
+// WriteLines writes a slice of strings to a file, each string being written to a new line.
+func WriteLines(filename string, lines []string) error {
+	file, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("failed to create file: %w", err)
+	}
+	defer file.Close()
+
+	for _, line := range lines {
+		_, err := file.WriteString(line + "\n")
+		if err != nil {
+			return fmt.Errorf("failed to write to file: %w", err)
+		}
+	}
+	return nil
+}
+
+// ParseKeyValue parses a key-value pair from a string and returns the key and value.
+func ParseKeyValue(line, ext string) (string, string) {
+	if ext == ".env" {
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) == 2 {
+			return strings.TrimSpace(parts[0]), strings.TrimSpace(parts[1])
+		}
+		return strings.TrimSpace(parts[0]), ""
+	} else if ext == ".out" {
+		parts := strings.Fields(line)
```

Comment:
we cant use this . what if the value has multiple spaces in it . then this will not work properly . we should split with first space and send the rest as value

Reply:
Done.